src/hera/workflows/io/_io_mixins.py

```diff
@@ -165,7 +202,10 @@ def _get_output(cls, field_name: str) -> Union[Artifact, Parameter]:
         return Parameter(name=field_name, default=default)  # type: ignore
 
     def _get_as_output(self) -> List[Union[Artifact, Parameter]]:
```

Comment:
Note @samj1912 this function is new but was accidentally added to the IO refactor (but was unused in that commit). TBD how/if we should change from "value from" `parameter` to `expression`.

Reply:
Confirmed that valueFrom->parameter and valueFrom->expression are functionally equivalent (and both work for the special `.result` output) using the following yaml:

```yaml
      outputs:
        parameters:
          - name: value
            valueFrom:
              parameter: '{{tasks.whalesay.outputs.result}}'
```

```yaml
      outputs:
        parameters:
          - name: value
            valueFrom:
              expression: 'tasks.whalesay.outputs.result'
```

Where they differ would be for garbage references and how the error show up in the UI, where the valueFrom->parameter makes a bit more sense:
```yaml
      outputs:
        parameters:
          - name: value
            valueFrom:
              parameter: '{{garbage-parameter}}'
```
![image](https://github.com/argoproj-labs/hera/assets/17798778/2ea4d156-d587-450f-a745-a63e5ee4e32a)

```yaml
      outputs:
        parameters:
          - name: value
            valueFrom:
              expression: "garbage-expression"
```
![image](https://github.com/argoproj-labs/hera/assets/17798778/aa81ed30-24d3-4d4f-9265-12a17a0c45e1)

Neither actually resolves the reference when submitting the workflow, and only fails when running 😐 This shouldn't be a problem because we're constructing the `valueFrom`.